tests/python/test_ad_tensor.py

```diff
@@ -0,0 +1,34 @@
+import functools
+
+import numpy as np
+import pytest
+
+import taichi as ti
+from tests import test_utils
+
+has_autograd = False
+
+try:
+    import autograd.numpy as np
+    from autograd import grad
+
+    has_autograd = True
+except:
+    pass
+
+
```

Comment:
Should we add a decorator similar to the one used in `test_ad_basics.py` at this location?

```suggestion
def if_has_autograd(func):
    # functools.wraps is nececssary for pytest parametrization to work
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        if has_autograd:
            func(*args, **kwargs)

    return wrapper
    
    
    @if_has_autograd
```